{
    "cve_id": "CVE-2023-6597",
    "cve_description": "An issue was found in the CPython `tempfile.TemporaryDirectory` class affecting versions 3.12.1, 3.11.7, 3.10.13, 3.9.18, and 3.8.18 and prior.\n\nThe tempfile.TemporaryDirectory class would dereference symlinks during cleanup of permissions-related errors. This means users which can run privileged programs are potentially able to modify permissions of files referenced by symlinks in some circumstances.\n",
    "cve_publish_date": "2024-03-19",
    "cwe_id": "NVD-CWE-noinfo",
    "commit_message": "gh-91133: tempfile.TemporaryDirectory: fix symlink bug in cleanup (GH-99930)\n\nCo-authored-by: Serhiy Storchaka <storchaka@gmail.com>",
    "type_of_change": "Modification",
    "filename_of_changes": "tempfile.py",
    "code_language": "Python",
    "number_of_lines_added_for_mitigation": "18",
    "number_of_lines_deleted_vulnerable_to_cve": "9",
    "vulnerable_lines": [
        "// Line_Reference 879:                 def resetperms(path):",
        "// Line_Reference 880:                     try:",
        "// Line_Reference 881:                         _os.chflags(path, 0)",
        "// Line_Reference 882:                     except AttributeError:",
        "// Line_Reference 883:                         pass",
        "// Line_Reference 884:                     _os.chmod(path, 0o700)",
        "// Line_Reference 885: ",
        "// Line_Reference 888:                         resetperms(_os.path.dirname(path))",
        "// Line_Reference 889:                     resetperms(path)"
    ]
}